server/src/main/java/io/crate/role/Roles.java

```diff
@@ -25,10 +25,33 @@
 
 import org.jetbrains.annotations.Nullable;
 
+import io.crate.common.FourFunction;
 import io.crate.metadata.pgcatalog.OidHash;
 
 public interface Roles {
 
+    FourFunction<Role, Privilege.Type, Privilege.Clazz, Object, Boolean> HAS_PRIVILEGE_FUNCTION = (r, t, c, o) ->
```

Comment:
Could we make those regular methods instead of function constants?

Reply:
I chose this, to have a common logic for the recursive check, and then apply a function that actually resolves the privilege for each case. What is your suggestion? (sorry I didn't get the "regular methods").